spanner/spannertest/integration_test.go

```diff
@@ -1145,6 +1160,31 @@ func TestIntegration_ReadsAndQueries(t *testing.T) {
 				{"Daniel"},
 			},
 		},
+		{
+			`SELECT JoinC.x, JoinD.z FROM JoinC JOIN JoinD USING (x) WHERE z != 'n' ORDER BY x`,
+			nil,
+			[][]interface{}{
+				{int64(2), "k"},
+				{int64(3), "m"},
+				{int64(3), "m"},
+			},
+		},
+		{
+			`SELECT * FROM JoinG RIGHT OUTER JOIN JoinD USING (x) WHERE JoinG.x IS NULL ORDER BY x, z, a, b`,
+			nil,
+			[][]interface{}{
+				{int64(2), nil, nil, "k"},
+				{int64(4), nil, nil, "p"},
+			},
+		},
+		{
+			`SELECT EpisodeID, offset, Cool FROM Episodes AS e, UNNEST(e.StaffIDs) as id WITH OFFSET as offset JOIN Staff AS s ON id = s.ID LEFT JOIN Staff as taller ON s.Height < taller.Height WHERE taller.ID IS NULL ORDER BY offset`,
+			nil,
+			[][]interface{}{
+				{int64(3), int64(0), true},
+				{int64(4), int64(3), true},
+			},
+		},
```

Comment:
This (or somewhere else, but I think this is the best place) could use some more test cases, especially for the base cases of what this PR introduces. So:
1. Simple `UNNEST` clauses, with and without `WITH OFFSET` (and with and without alias for the offset).
2. Simple Cartesian products using both 2 and 3 tables, and using both all tables filled with data and scenarios where at least one table does not contain data.